flocker/ca/_ca.py

```diff
@@ -231,14 +231,81 @@ def write_credential_files(self, key_filename, certificate_filename):
             os.umask(original_umask)
 
 
-class NodeCredential(FlockerCredential):
+class UserCredential(PRecord):
+    """
+    A certificate for an API user, signed by a supplied certificate
+    authority.
+
+    :ivar FlockerCredential credential: The certificate and key pair
+        credential object.
+    :ivar bytes username: A username.
+    """
+    credential = field(mandatory=True)
+    username = field(mandatory=True, initial=None)
+
+    @classmethod
+    def from_path(cls, path, username):
+        """
+        Load a node certificate from a specified path.
+        """
+        key_filename = b"{user}.key".format(user=username)
+        cert_filename = b"{user}.crt".format(user=username)
+        keypair, certificate = load_certificate_from_path(
+            path, key_filename, cert_filename
+        )
+        credential = FlockerCredential(
+            path=path, keypair=keypair, certificate=certificate)
+        return cls(credential=credential, username=username)
+
+    @classmethod
+    def initialize(cls, path, authority, username):
+        """
+        Generate a certificate signed by the supplied root certificate.
+
+        :param FilePath path: Directory where the certificate will be stored.
+        :param CertificateAuthority authority: The certificate authority with
+            which this certificate will be signed.
+        :param bytes username: The username to be included in the certificate.
+        """
+        key_filename = b"{user}.key".format(user=username)
+        cert_filename = b"{user}.crt".format(user=username)
+        # The common name for the node certificate.
+        name = b"user-{user}".format(user=username)
+        # The organizational unit is set to the common name of the
+        # authority, which in our case is a byte string identifying
+        # the cluster.
+        organizational_unit = authority.credential.certificate.getSubject().CN
+        dn = DistinguishedName(
+            commonName=name, organizationalUnitName=organizational_unit
+        )
+        keypair = flocker_keypair()
+        request = keypair.keypair.requestObject(dn)
+        serial = os.urandom(16).encode(b"hex")
+        serial = int(serial, 16)
+        cert = authority.credential.keypair.keypair.signRequestObject(
+            authority.credential.certificate.getSubject(), request,
+            serial, EXPIRY_20_YEARS, 'sha256'
+        )
+        credential = FlockerCredential(
+            path=path, keypair=keypair, certificate=cert
+        )
+        instance = cls(credential=credential, username=username)
+        instance.credential.write_credential_files(
+            key_filename, cert_filename)
```

Comment:
You could use `credential` to do this (and perhaps before instantiating `cls` so that the instance is always valid instead of maybe referring to files that don't actually exist - as could happen if `write_credential_files` raises an exception because it can't do the write for some reason).